followthemoney/proxy.py

```diff
@@ -209,21 +209,27 @@ def unsafe_add(
         """A version of `add()` to be used only in type-checking code. This accepts
         only a single value, and performs input cleaning on the premise that the
         value is already valid unicode."""
+        original_value = value
+
         if not cleaned and value is not None:
             value = prop.type.clean_text(value, fuzzy=fuzzy, format=format, proxy=self)
+
+        if original_value is not None and value is None:
+            log.warning(f"[{prop.name}] Removed unclean value \"{original_value}\"")
```

Comment:
Note to myself: This is a very simple way to add some logging when unclean property values are removed.

It logs something like this:

```
followthemoney.proxy [WARNING] [phone] Removed unclean value "(567) 992-0000"
```

This isn’t perfect for debugging, it only mentions the FtM property name and the invalid value.

It would be helpful if we could also display the primary key of the entity that is currently being mapped so mapping developers can easily find the relevant source records. Two problems with that though:

1. It would require a few changes to the `PropertyMapping` class as it is currently not aware of the entity mapping it is part of. Also would require changes to `EntityProxy.unsafe_add` to somehow indicate if a value passed cleaning or not.

2. The entity key may not be known at this time, because the key properties haven’t been mapped yet or the cleaned value is itself part of a key property.

Reply:
As a low-hanging fruit, we could at least also log the schema name and property type of the affected property:

```
followthemoney.proxy [WARNING] Removed unclean value "(567) 992-0000" from property "Person:phone" of type "phone".
```

Reply:
@brrttwrks @uhhhuh @Okssana Any other information that would be helpful when debugging unclean values in mappings?